File: in_Java/0160 Intersection of Two Linked Lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        ListNode a = headA, b = headB;
        int lena = 0, lenb = 0;
        while(a != null){
            lena++;
            a = a.next;
        }
        while(b != null){
            lenb++;
            b = b.next;
        }
        int d;
        if(lena > lenb){
            d = lena-lenb;
            while(d-- != 0){
                headA = headA.next;
            }
        }
        if(lena < lenb){
            d = lenb-lena;
            while(d-- != 0){
                headB = headB.next;
            }
        }
        while(headA != null && headB != null){
            if(headA == headB) return headA;
            headA = headA.next;
            headB = headB.next;
        }
        return null;
    }
}

File: in_Python/0160 Intersection of Two Linked Lists.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def getIntersectionNode(self, headA, headB):
        """
        :type head1, head1: ListNode
        :rtype: ListNode
        """
        la = 0
        lb = 0
        h1 = headA
        h2 = headB
        while h1 != None:
            h1 = h1.next
            la += 1
        while h2 != None:
            h2 = h2.next
            lb += 1
        d = la - lb
        if d > 0:
            while d != 0:
                headA = headA.next
                d -= 1
        else:
            while d != 0:
                headB = headB.next
                d += 1
        while headA != headB:
            headA = headA.next
            headB = headB.next
        return headA